### PR TITLE
[MAINT] monitor duration of doc build

### LINKED ISSUE #4385: update `maint_tools/check_gha_workflow.py` to also monitor the doc build on main
https://github.com/nilearn/nilearn/blob/main/maint_tools/check_gha_workflow.py

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes #4385

Changes proposed in this pull request:

- update script and GH workflow to plot duration of doc builds

### CODE REVIEW

**Comment by Remi-Gau** [comment]
output example

doc_runs_timing

**Review by man-shu — APPROVED** [review]
LGTM!

File: maint_tools/check_gha_workflow.py [hunk 1]
```diff
@@ -127,10 +106,72 @@ def main(args=sys.argv) -> None:
...
+    print(df)
+
+    plot_doc_job_durations(df, output_file)
 
 
-def plot_job_durations(df: pd.DataFrame) -> None:
+def update_tsv(
+    args, update: bool, output_file: Path, workflow_id: str
+) -> None:
+    """Update TSV containing run time of every workflow."""
```
Comment (bthirion) [inline maint_tools/check_gha_workflow.py:143]: If the function is not used elsewhere, should be `_update_tsv`, right ?

File: maint_tools/check_gha_workflow.py [hunk 2]
```diff
@@ -127,10 +106,72 @@ def main(args=sys.argv) -> None:
...
+            print(f" found {len(runs)} runs")
+            jobs_data = update_jobs_data(jobs_data, runs, auth)
+        else:
+            break
+
+    df = pd.DataFrame(jobs_data)
+    df.to_csv(output_file, sep="\t", index=False)
+
+
+def plot_test_job_durations(df: pd.DataFrame, output_file: Path) -> None:
```
Comment (bthirion) [inline maint_tools/check_gha_workflow.py:174]: Same for other functions.

**Review by bthirion — COMMENTED** [review]
LGTM overall, but need to standardize the syntax for internal functions.

File: maint_tools/check_gha_workflow.py [hunk 3]
```diff
@@ -61,13 +61,9 @@
...
 TOKEN_FILE = Path("/home/remi/Documents/tokens/gh_read_repo_for_orga.txt")
 
-# can be found out at
-# "https://api.github.com/repos/{USER}/{REPO}/actions/workflows"
-WORKFLOW_ID = "71549417"
-
 BRANCH = "main"
 
-INCLUDE_FAILED_RUNS = True
+INCLUDE_FAILED_RUNS = False
```
Comment (bthirion) [inline maint_tools/check_gha_workflow.py:66]: I'm wondering what this entails... Can you explain the rationale ?

**Review by bthirion — COMMENTED** [review]
Thx !

File: maint_tools/check_gha_workflow.py (hunk 1 above)
Comment (Remi-Gau) [reply maint_tools/check_gha_workflow.py:143]: Good catch. Will fix.

File: maint_tools/check_gha_workflow.py (hunk 3 above)
Comment (Remi-Gau) [reply maint_tools/check_gha_workflow.py:66]: Will add some doc to explain this.

File: maint_tools/check_gha_workflow.py (hunk 2 above)
Comment (Remi-Gau) [reply maint_tools/check_gha_workflow.py]: done

File: maint_tools/check_gha_workflow.py (hunk 1 above)
Comment (Remi-Gau) [reply maint_tools/check_gha_workflow.py:145]: done

File: maint_tools/check_gha_workflow.py (hunk 3 above)
Comment (Remi-Gau) [reply maint_tools/check_gha_workflow.py:68]: Added some explanation.
@bthirion is this clearer?

Comment (bthirion) [reply maint_tools/check_gha_workflow.py:68]: OK

**Review by bthirion — APPROVED** [review]
LGTM, thx !

**Pull request merged**